components/nexus-repository/src/main/java/org/sonatype/nexus/repository/search/SearchFacetImpl.java

```diff
@@ -98,11 +123,10 @@ protected void put(final EntityId componentId) {
         }
         assets = Lists.newArrayList(tx.browseAssets(component));
       }
-      String json = JsonUtils.merge(componentMetadata(component, assets), JsonUtils.from(additional));
-      searchService.put(getRepository(), EntityHelper.id(component).toString(), json);
+      put(component, assets);
     }
-    catch (IOException e) {
-      throw Throwables.propagate(e);
```

Comment:
Good question: as I see, IOEx is thrown only if some problem happens during JSON operations (from/merge). But, if IOException does happens during managing Component and Asset attributes, it might/should be actually considered as a bug (ie. invalid value got into their attributes?)

Anyway, am returning the catch for now, to not fail rebuild if one document is broken for now, and let's discuss.